pkgs/build-support/flatten-references-graph/src/flatten_references_graph/flatten_references_graph.py

```diff
@@ -0,0 +1,45 @@
+from toolz import curried as tlz
+
+from .lib import (
+    flatten,
+    over,
+    references_graph_to_igraph
+)
+
+from .pipe import pipe
+
+MAX_LAYERS = 127
+
+
+def create_list_of_lists_of_strings(deeply_nested_lists_or_dicts_of_graphs):
+    list_of_graphs = flatten(deeply_nested_lists_or_dicts_of_graphs)
+
+    return list(
+        filter(
+            # remove empty layers
+            lambda xs: len(xs) > 0,
+            tlz.map(
+                lambda g: g.vs["name"],
+                list_of_graphs
+            )
+        )
+    )
+
+
+def flatten_references_graph(references_graph, pipeline, exclude_paths=None):
+    if exclude_paths is not None:
```

Comment:
ditto, that conditional would be gone if we were to have `exclude_paths` as an empty list. Less branches on code, less chances of failure.

Reply:
actually the conditional would still be there, it would simply change to `isEmpty(exclude_paths)` since if there is nothing to exclude, there is no point iterating the graph and making a copy of it

unless you can propose an algo which is be driven by `exclude_paths` and boils down to a no-op if it's empty